lib/ProductOpener/APIProductWrite.pm

```diff
@@ -465,21 +482,82 @@ sub write_product_api ($request_ref) {
 			);
 		}
 		else {
-			# Update the product
-			update_product_fields($request_ref, $product_ref, $response_ref);
 
-			# Process the product data
-			analyze_and_enrich_product_data($product_ref, $response_ref);
+			# Change of code
+			if (    (defined $request_body_ref->{product}{code})
+				and ($product_ref->{code} ne $request_body_ref->{product}{code}))
```

Comment:
I'm a bit disappointed that we have to do those checks here and in product_jqm_multilingual… could'nt we factor this out ?